transforms/language/text_encoder/python/src/text_encoder_transform.py

```diff
@@ -0,0 +1,137 @@
+from data_processing.utils import CLIArgumentProvider, TransformUtils
+from sentence_transformers import SentenceTransformer
+
+
+short_name = "text_encoder"
+cli_prefix = f"{short_name}_"
+model_name_key = "model_name"
+content_column_name_key = "content_column_name"
+output_embeddings_column_name_key = "output_embeddings_column_name"
+model_name_cli_param = f"{cli_prefix}{model_name_key}"
+content_column_name_cli_param = f"{cli_prefix}{content_column_name_key}"
+output_embeddings_column_name_cli_param = (
+    f"{cli_prefix}{output_embeddings_column_name_key}"
+)
+
+default_model_name = "BAAI/bge-small-en-v1.5"
+default_content_column_name = "contents"
+default_output_embeddings_column_name = "embeddings"
+
+
+class TextEncoderTransform(AbstractTableTransform):
+    """
+    Implements a simple copy of a pyarrow Table.
+    """
+
+    def __init__(self, config: dict[str, Any]):
+        """ """
+        # Make sure that the param name corresponds to the name used in apply_input_params method
+        # of TextEncoderTransform class
+        super().__init__(config)
+        from data_processing.utils import get_logger
+
+        self.logger = get_logger(__name__)
+
+        self.model_name = config.get(model_name_key, default_model_name)
+        self.content_column_name = config.get(
+            content_column_name_key, default_content_column_name
+        )
+        self.output_embeddings_column_name = config.get(
+            output_embeddings_column_name_key, default_output_embeddings_column_name
+        )
+
+        self.model = SentenceTransformer(self.model_name)
+
+    def transform(
+        self, table: pa.Table, file_name: str = None
+    ) -> tuple[list[pa.Table], dict[str, Any]]:
+        """ """
+        self.logger.debug(f"Transforming one table with {len(table)} rows")
+
+        # make sure that the content column exists
+        TransformUtils.validate_columns(
+            table=table, required=[self.content_column_name]
+        )
+
+        embeddings = list(
+            map(
+                lambda x: self.model.encode(x, normalize_embeddings=True),
+                table[self.content_column_name].to_pylist(),
+            ),
+        )
+        result = TransformUtils.add_column(
+            table=table, name=self.output_embeddings_column_name, content=embeddings
+        )
+
+        metadata = {"nfiles": 1, "nrows": len(result)}
+        return [result], metadata
+
+
+class TextEncoderTransformConfiguration(TransformConfiguration):
+    """
+    Provides support for configuring and using the associated Transform class include
+    configuration with CLI args.
+    """
+
+    def __init__(self):
+        super().__init__(
+            name=short_name,
+            transform_class=TextEncoderTransform,
+            # remove_from_metadata=[pwd_key],
+        )
+        from data_processing.utils import get_logger
+
+        self.logger = get_logger(__name__)
```

Comment:
same comment on duplicate logging as for chunker.  Recommend using `__name__ + "cfg"` or some such to distinguish name from one used by the transform itself.